ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -242,40 +242,103 @@ module Json = struct
             ; ("func_name_suffix", `String (suffix_of_type t))
             ]
 
-  let of_params params =
-    let name_internal name =
-      let name = name |> snake_to_camel |> String.uncapitalize_ascii in
-      match name with "type" -> "typeKey" | "interface" -> "inter" | _ -> name
+  let get_last l = match List.rev l with h :: _ -> Some h | _ -> None
+
+  let group_params msg =
+    let groups params =
+      let params =
+        List.map
+          (fun p -> (published_release_for_param p.param_release.internal, p))
+          params
+      in
+      let uniq_sorted_rels =
+        List.map fst params
+        |> Listext.List.setify
+        |> List.fast_sort compare_versions
+      in
+      List.map
+        (fun rel ->
+          let params =
+            List.filter_map
+              (fun (r, param) ->
+                match compare_versions r rel with
+                | n when n > 0 ->
+                    None
+                | _ ->
+                    Some param
+              )
+              params
+          in
+          (params, rel)
+        )
+        uniq_sorted_rels
     in
-    let of_param param =
-      let suffix_of_type = suffix_of_type param.param_type in
-      let t, _e = string_of_ty_with_enums param.param_type in
-      let name = param.param_name in
+    let only_session_group =
       [
-        ("is_session_id", `Bool (name = "session_id"))
-      ; ("type", `String t)
-      ; ("name", `String name)
-      ; ("name_internal", `String (name_internal name))
-      ; ("doc", `String param.param_doc)
-      ; ("func_name_suffix", `String suffix_of_type)
+        ( [session_id]
+        , published_release_for_param session_id.param_release.internal
+        )
       ]
     in
-    (* We use ',' to seprate params in Go function, we should ignore ',' before first param,
-       for example `func(a type1, b type2)` is wanted rather than `func(, a type1, b type2)`.
-    *)
-    let add_first = function
-      | head :: rest ->
-          let head = `O (("first", `Bool true) :: of_param head) in
-          let rest =
-            List.map
-              (fun item -> `O (("first", `Bool false) :: of_param item))
-              rest
-          in
-          head :: rest
+    let add_session_id = function
       | [] ->
-          []
+          only_session_group
+      | groups ->
+          List.map (fun (params, rel) -> (session_id :: params, rel)) groups
+    in
+    let groups =
+      if msg.msg_session then
+        msg.msg_params |> groups |> add_session_id
+      else
+        groups msg.msg_params
+    in
+    (* The bool label in the tuple below is tagged to distinguish whether it is the latest parameters group.
+       If it's the latest parameters group, then we should not add the number of parameters to the method's name.
+    *)
+    match get_last groups with
+    | Some ([], rel) ->
+        [(true, [], rel)]
+    | Some ([x], rel)
```

Comment:
This case seems too confusing. 
As discussed, this is for the function like `Session.foo()` in which case we don't want to generate `Session.foo0()`.
But I think ``Session.foo0()` is not so bad, e.g. when there will be a `Session.foo(x)`.

Reply:
I agree with that.